### ISSUE TITLE
Refactor lens profile handling

### ISSUE THREAD

**Issue by AdrianEddy** [gyroflow/gyroflow#742]
- [ ] Create a list of known camera brands, models and lenses. To collect the data use the current lens profile database and the lensfun database. Store it as json in `resources` and have it auto-update like lens profiles (also should be in [git](https://github.com/gyroflow/lens_profiles/))
- [ ] Create a list of compatible cameras, lens mounts and sensor sizes, also include crop factor
- [ ] Calibrator should enforce to select correct camera from the list, with the ability to select "Other" and type manually
- [ ] Lens profile search should be changed to a selector for camera brand, model and lens model. This way it can also automatically recommend compatible cameras (for example all Sony full frame cameras are compatible)
- [ ] Include the LensFun database in the selectors #150 
- [ ] Implement a review process which allows to toggle between all submitted lens profiles for a given camera setup. This will make it easy to remove bad ones
- [ ] All selectors (in calibrator and in main window) should be pre-populated with the metadata from video file as much as we can
- [ ] Calibrator: Don't allow to submit lens profiles without a clear distinction of the setup (eg. mirrorless camera model without lens params, zoom lens without specific focal length info, no camera model or brand name, etc)

**Comment by SWAYAMKHANDURI** [comment]
Hey there @AdrianEddy regarding the first task : camera brands, models and lenses are already present in resources can you assist me here 
PS: I might be mistaken cuz just starting out in OS Community 😁 
![image](https://github.com/gyroflow/gyroflow/assets/63970499/7f550af9-fb15-44dd-97be-9a27d2917b80)

**Comment by AdrianEddy** [comment]
The point is to make a list of real brands, models, lens models in order to standardize and cleanup the collection.

What's already in lens profiles are just strings and they are all over the place. We need a solid list of possible values and make people choose from that list, instead of typing manually because it leads to many different strings when users type with mistakes.

That list should be independent of the lens profile and should be the base for the comboboxes/selectors in the UI. Then we should go through all lens profiles and make sure these strings match exactly and clean it all up

**Comment by nicefacer** [comment]
@AdrianEddy can I join to fix this?

**Comment by AdrianEddy** [comment]
sure, feel free to join

**Comment by AdrianEddy** [comment]
I recommend collaborating, there are plenty of tasks for this issue

**Comment by AdrianEddy** [comment]
I created a thread on Discord for discussion, I'll also be there to guide you https://discord.com/channels/797044698682228736/1164959165609934959 (#dev-software -> Lens profile refactor)

**Comment by nicefacer** [comment]
Hey @SWAYAMKHANDURI can we collab on this? waiting for your thought.

**Comment by nicefacer** [comment]
> I created a thread on Discord for discussion, I'll also be there to guide you https://discord.com/channels/797044698682228736/1164959165609934959 (#dev-software -> Lens profile refactor)

Unfortunately, I don have access to the server. can you pls, invite, then I will find the channel above?

**Comment by algora-pbc** [comment]
hey guys, if you collaborate you can split the bounty using the `/split @username` command after the `/claim #742` command in the PR -- cheers, algora team

**Comment by AdrianEddy** [comment]
> Hello @AdrianEddy, Unfortunately, I don have access to the server. can you pls, invite, then I will find the channel above?

https://discord.com/invite/BBJ2UVAr2D

**Comment by AdrianEddy** [comment]
#150 may also be part of this and it has a separate bounty. It's not strictly necessary to be done together with this issue, but is also an option

**Comment by SWAYAMKHANDURI** [comment]
@nicefacer yeah sure looking forward to collaborating with you 😊

**Comment by mihir1739** [comment]
Is this issue solved already?

**Comment by AdrianEddy** [comment]
No, nothing has been done towards resolving it

**Comment by mihir1739** [comment]
Okay. I'll work on it then
/attempt #742

**Comment by Rushhaabhhh** [comment]
@AdrianEddy -
Is this issue open to work and claim the bounty ?

**Comment by AdrianEddy** [comment]
yes

**Comment by Rushhaabhhh** [comment]
@AdrianEddy  - Okay so I'll be working on this issue, thankyou.
/attempt #742

### Implementation Plan
1. Integrate the LensFun database and existing profiles into JSON format in resources with auto-updating functionality.
2. Replace the lens profile search with brand, model, and lens selectors, pre-populated using video metadata.
3. Enforce validation to ensure all necessary metadata is provided, with an "Other" option for manual input.
4. Add a review feature to toggle and manage submitted lens profiles.

Also I’ll need guidance on the issue to work efficiently and as per requirements.

**Comment by Rushhaabhhh** [comment]
@AdrianEddy - 

I have a few questions regarding the bounty : 

1. What are the specific conditions or requirements to qualify for the bounty ?


2. When is the project considered complete and eligible for the bounty ? Are there any specific criteria or deliverables to meet ?


3. What are the detailed terms and conditions of the bounty ?


4. Are there any additional steps or documentation required to claim the bounty ?


5. Does the process of solving this issue also cover the scope and bounty of issue #120, given that the work on this project involves elements related to it ?

6. Is there a deadline for submission ?


Please let me know if there’s anything else I should be aware of regarding this issue.

**Comment by Rushhaabhhh** [comment]
@AdrianEddy Any updates on the doubts ?

**Comment by tmahmood** [comment]
Hey @AdrianEddy 

Is this task available? Maybe I can help with this?

**Comment by shubham-shinde-442** [comment]
Hi @AdrianEddy , I’d like to attempt this and have some experience in QML / Qt software development. Is it still available ?

**Comment by debaa98** [comment]
hey @AdrianEddy is this bounty still active ?

**Comment by AdrianEddy** [comment]
yes, no one provided any code for it yet

**Comment by nicefacer** [comment]
@AdrianEddy did you disconnect it from algora? i am currently testing what i done so, far on this.

**Comment by AdrianEddy** [comment]
no, it's still connected to Algora

<img width="1120" height="444" alt="Image" src="https://github.com/user-attachments/assets/2f3051b5-0499-4ac0-b040-19d9d820aa62" />

**Comment by nicefacer** [comment]
sent a PR @AdrianEddy

**Comment by nicefacer** [comment]
Hello, @AdrianEddy , not even a comment on the pull request? has someone solved to the issue?

**Comment by AdrianEddy** [comment]
I just didn't have the time to review it yet, but in any case that PR is only for the first point in this issue

**Comment by nicefacer** [comment]
pls, give more details. what is the 2nd? i want to fix everything at the same time

**Comment by AdrianEddy** [comment]
so your PR only takes care of this
```
Create a list of known camera brands, models and lenses. To collect the data use the current lens profile database and the lensfun database. Store it as json in resources and have it auto-update like lens profiles (also should be in [git](https://github.com/gyroflow/lens_profiles/))
```
 (it's also incomplete, because you didn't use the existing database to collect brands/models/lenses)

and all the remaining points are not implemented at all:
```
- Create a list of compatible cameras, lens mounts and sensor sizes, also include crop factor
- Calibrator should enforce to select correct camera from the list, with the ability to select "Other" and type manually
- Lens profile search should be changed to a selector for camera brand, model and lens model. This way it can also automatically recommend compatible cameras (for example all Sony full frame cameras are compatible)
- Include the LensFun database in the selectors https://github.com/gyroflow/gyroflow/issues/150
- Implement a review process which allows to toggle between all submitted lens profiles for a given camera setup. This will make it easy to remove bad ones
- All selectors (in calibrator and in main window) should be pre-populated with the metadata from video file as much as we can
- Calibrator: Don't allow to submit lens profiles without a clear distinction of the setup (eg. mirrorless camera model without lens params, zoom lens without specific focal length info, no camera model or brand name, etc)
```

**Comment by nicefacer** [comment]
thank you @AdrianEddy , kindly clarify these:
What fields do you want in the camera database for sensor size/crop—e.g., physical sensor dims, diagonal, crop factor per mount/body, flange distance?

For LensFun ingestion, do we prefer embedding parsed data into camera_database.json or merging at runtime from the lensfun XMLs?

Should the brand/model/lens selector drive lens profile auto-selection only, or also constrain calibration submissions (block save/upload unless selection is made)?

**Comment by nicefacer** [comment]
Hey, @AdrianEddy , compliment of the session. still waiting for your answer so, i can fix this issue this week

**Comment by bobbiejaxn** [comment]
Hi @AdrianEddy, I'm interested in contributing to this bounty. I noticed PR #1118 has been open for ~2 months with no reviews yet.

Before diving in, I wanted to check:
1. Is PR #1118 still being considered, or should fresh approaches start from scratch?
2. For the LensFun XML database import requirement — would you prefer using the `quick-xml` crate or `xml-rs`? I notice the existing codebase doesn't currently have an XML parser dependency.
3. Would you prefer incremental PRs (e.g., camera/lens database first, then UI, then validation) or one comprehensive PR?

I've already submitted PR #1150 for issue #150 (Lensfun database import) which covers the lens profile data side. Happy to collaborate or take a different direction based on your feedback.

Thanks for the great project!

**Comment by ProspectOre** [comment]
/attempt #742

Implementing all eight checklist items.

- Camera/lens database (~2.5k cameras, ~6.7k lenses) merged from existing gyroflow lens profiles and LensFun, shipped as JSON in resources, auto-updated from the lens_profiles release.
- Compatibility info: mount, sensor size, crop factor; compatible-camera lookup based on shared mount + sensor class.
- Main window lens search replaced with brand/model/lens selectors; surfaces compatible cameras.
- Calibrator: same selectors with "Other" for manual entry.
- LensFun ingested in the build script.
- Review UI: cycle through submitted profiles for a setup, hide bad ones locally.
- All selectors pre-populate from video metadata when present.
- Calibrator validation blocks submission without a clear setup distinction (missing brand/model, unknown camera, missing lens for non-fixed cameras, zoom without focal length).

Two PRs incoming:
- lens_profiles: build script + release workflow + brand normalization across existing profile JSONs.
- gyroflow: loader, controller API, selector UI, review UI, calibrator validation.

**Comment by bobbiejaxn** [comment]
Hi @AdrianEddy and @ProspectOre — I'm working on **sub-task 8** from this issue: calibrator submission validation (blocking profiles without camera brand/model, mirrorless cameras without lens params, and zoom lenses without focal length info).

This is focused, complementary work that hasn't been covered by any existing PR. I plan to implement validation logic in the Rust backend and UI error states in LensCalibrate.qml that prevent submission until all required fields are present.

PR #1154 covers sub-tasks 1–7 comprehensively. My submission validation work could merge independently or alongside it. Happy to coordinate if needed.

/cc @ProspectOre — interested in splitting the bounty if our changes can coexist?

**Comment by aimqwest** [comment]
/attempt
AIMQWEST submission: preparing a focused refactor for lens profile handling with backward-compatible behavior and test coverage.

**Comment by wangedmund77-cmyk** [comment]
/attempt #742

I’d like to take this on, assuming the bounty is still active.

My plan is to first map the current lens profile database, LensFun import path, and profile search/calibrator flows, then keep the implementation reviewable in scoped steps:

- generate structured camera / model / lens metadata JSON from the existing profiles plus LensFun data
- add or extend the auto-update script so the metadata stays in sync with lens profiles
- update the lens profile search flow toward brand / camera model / lens selectors while preserving “Other” manual entry
- add validation so submitted calibration profiles include enough camera/lens/focal metadata to be useful
- include fixtures/tests around matching and metadata generation

If the review workflow is too large for the first PR, I’ll split that into a follow-up and make the first PR independently useful.

**Comment by RDawg29** [comment]
/attempt #742

I reviewed the active related PRs (#1118, #1154, #1156, #1158) and I do not want to duplicate work already in flight.

My plan is to start with a focused audit against the issue checklist and the companion lens profile database work, then open a small PR only for a concrete remaining gap that can coexist cleanly with the leading implementation. Likely areas I will check first:

- calibrator/profile submission validation edge cases
- review-flow behavior for multiple submitted profiles
- metadata prefill and alias-normalization cases
- missing tests or demo coverage that would make the maintainer review easier

If the maintainers prefer this to land as review/test coverage or as a follow-up against the main implementation, I am happy to coordinate and use `/split` if appropriate.

**Comment by jamilahmadzai** [comment]
/attempt #742

I am taking a focused, reviewable pass on the app-side lens profile flow on top of current master:

- build a camera/lens registry from the loaded Gyroflow profile database, with an optional `camera_registry.json` release asset hook for future catalog data
- expose brand/model/lens selectors to the lens profile search and calibrator flows
- prefill selectors from video metadata when a camera identifier is available
- keep profile review local with previous/next controls and local hide state for bad profiles
- block uploads of ambiguous calibration profiles while still allowing the explicit Other/manual path
- make the profile updater select release assets by name so adding registry assets does not break profile downloads

I will include focused core tests for registry construction and selector-based search behavior.

**Comment by Ishant5436** [comment]
/attempt #742

Plan: submit a distinct lens profile schema guard to help enforce bullet 8 of the refactor constraints. This ensures the Calibrator logic does not ingest profiles lacking camera setups, focal length data, or brand metadata.

Scope: A standalone, dependency-free python validator that can be hooked into CI or the local builder to strictly validate incoming JSON schemas. Submitted via focused PR with /claim #742.

**Comment by helpinghandschitown-netizen** [comment]
/attempt #742

I submitted PR #1195 as an independent attempt covering the full eight-item checklist. The branch incorporates the four commits from #1154 with their original author metadata preserved and cites the companion gyroflow/lens_profiles#31 work. @ProspectOre is not a collaborator on PR #1195 and does not endorse it.

My additional work focuses on review-flow correctness and regression hardening: request-scoped async search results, serializing asynchronous `.gyroflow` preset selections, reliable hidden-profile identity, selected-profile navigation alignment, empty-lens validation, and stale detected-camera reset behavior. The PR includes test evidence and remains a draft until the Windows build/demo is attached.

**Comment by ProspectOre** [comment]
@AdrianEddy @algora-pbc Before #1195 is registered as an independent claim, I’d like to clarify its relationship to my existing submission.

#1195 replays the first four commits from my #1154, with authorship preserved. Those commits establish the implementation across nine files. Comparing the branch after those four imported commits with the three subsequent follow-up commits, my imported work accounts for approximately **99.8% of the changed-line volume when the generated camera database is included**, and approximately **91% when generated JSON is excluded** (2,052 of 2,258 non-generated added/deleted lines). The three subsequent commits by the #1195 author add 179 lines and delete 27 across four of those existing files. The submission also relies on my companion gyroflow/lens_profiles#31 work.

I’m not disputing the ability to build on open-source code. However, I did not agree to collaborate on or split this bounty, and #1195 is predominantly derivative of my submitted implementation. Please review #1154 and gyroflow/lens_profiles#31 as the original submission and clarify how attribution and bounty eligibility will be handled before registering or rewarding #1195 independently.

**Comment by ElvinGts** [comment]
I have prepared the aggregated camera/lens JSON database schema with crop factors, lens mount compatibility resolution, and metadata auto-detection for issue #742. Ready to submit PR! /claim

**Comment by valimikayilov** [comment]
/attempt #742

Submitted the implementation in #1212 and gyroflow/lens_profiles#37: a reproducible Gyroflow/Lensfun catalogue, shared selectors with metadata prefill, compatibility suggestions, review navigation, local hide/restore and upload validation. The application PR includes a short demo walkthrough, native build results and core/Qt test results. Lensfun supplies catalogue metadata; this does not add another distortion-model import implementation.

**Comment by Glastonburyk** [comment]
/attempt #742

Implementation plan: Add a canonical camera/lens metadata catalogue, expose it through the Rust/QML layers, replace free-form profile selection with structured camera/model/lens selectors while retaining an Other option, pre-populate from video metadata, tighten calibration validation, and add profile browsing/review support with focused tests.

**Comment by guwenly** [comment]
/attempt #742

Working on a complete checklist implementation (not a partial first-bullet PR). Plan:

1. **Canonical catalog JSON** — derive brands/models/lenses from the existing Gyroflow lens profile DB + LensFun; ship under `resources` with the same auto-update path as lens profiles; companion updates in `gyroflow/lens_profiles` as needed. Normalize messy free-text strings to stable values.
2. **Compatibility data** — mounts, sensor sizes, crop factors; recommend compatible cameras (e.g. same mount + sensor class / Sony FF peers).
3. **Selectors UI** — replace free-form lens profile search with brand → model → lens selectors in main window and calibrator; keep explicit **Other** + manual entry.
4. **Metadata prefill** — pre-populate selectors from video file metadata whenever present.
5. **Calibrator validation** — block submit/upload when setup is ambiguous (missing brand/model, zoom without focal length, mirrorless without lens params, etc.) while allowing the Other path.
6. **Review flow** — toggle/browse all submitted profiles for a given camera setup to spot/remove bad ones.
7. **LensFun in selectors** — include LensFun catalog metadata in the selectors (#150 overlap as appropriate).
8. **Demo + tests** — focused core tests for registry/search/validation; short demo video of the new selectors/validation/review in the PR; include `/claim #742` when ready.

I'll study existing open PRs so we don't re-ship incomplete work, and ask on Discord if any checklist item needs clarification before coding ambiguous UI behavior.

**Comment by ElvinGts** [comment]
/attempt #742

I have completed the aggregated camera/lens JSON database schema with full sensor coverage, crop factors, lens mount compatibility resolution, and video metadata auto-detection.

### Test Suite Results:
- All cameras contain brand, model, mount, sensor dimensions, and crop factors.
- Sony FX3 resolves compatibility with A7S III while correctly excluding APS-C FX30.
- Auto-detects GoPro HERO12 Black from video filename and EXIF tags.
- 5/5 unit tests passed (100% pass rate).

Ready to submit the PR!

**Comment by k4it00** [comment]
/attempt #742

**Implementation plan (staged):**

**Stage 1 — data layer (this PR):**
1. Generator script (Python, pinned deps) that merges the Gyroflow lens profile database metadata with the LensFun database (cameras.xml/lenses.xml: mounts, crop factors, model names) and emits `resources/cameras.json` with: brand → camera model → mount + crop factor + compatible lenses (in-library cross-referenced).
2. GitHub Action mirroring the lens-profiles auto-update flow (scheduled + manual dispatch) committing the refreshed JSON to the repo.
3. Includes LensFun coverage alongside Gyroflow profiles (foundations for item 5).

**Stage 2 — UI (follow-up PRs on the same issue):**
4. Calibrator camera selector enforcing list selection with an "Other/manual" escape hatch.
5. Lens profile search → hierarchical brand/model/lens selector with compatibility recommendation (e.g., all Sony full-frame bodies for a full-frame lens).
6. Metadata pre-population of selectors from video file metadata.

Will start with Stage 1 immediately — the JSON schema will be designed so the selectors can consume it directly.

**Comment by k4it00** [comment]
Withdrawing my /attempt — on closer inspection there are 13 open PRs already covering every checklist item (including LensFun import tooling and the camera database), so I don't think I can add differentiated value here. Good luck to everyone working on it!

**Comment by snchngny** [comment]
Is there a remaining, independently payable task under #742 that does not overlap with #1154? If so, please specify its scope, acceptance criteria, and reward. Would an AI-assisted implementation with reproducible tests be eligible?

**Comment by ElvinGts** [comment]
Hi @snchngny! I've already opened PR #1218 covering the core canonical camera/lens JSON database schema, physical sensor dimensions, and mount compatibility resolution.

If you're working on the UI selectors (LensCalibrate.qml / LensProfile.qml) or Lensfun integration, I'm completely open to collaborating and splitting the bounty via `@algora-pbc /split` as recommended by @AdrianEddy!

**Comment by snchngny** [comment]
Thanks for reaching out. The current #1218 diff adds the JSON database; is the loader and test code in another branch? A focused UI-selector task could be a useful split. Could we agree on the schema, exact UI scope, and bounty shares, and have Adrian confirm the split before implementation?

**Comment by ElvinGts** [comment]
@snchngny Thanks for following up!

To address your questions:

1. **Loader & Test Suite**: The database loader, schema validation, and test suite (which validates mounts, cameras, sensor dimensions, crop factors, and video EXIF matching with 5/5 tests passing) were developed and verified locally against the Gyroflow architecture. PR #1218 currently provides the canonical schema (`resources/camera_lens_db.json` v2.5.0). We can push the Rust/QML loader integration directly to #1218 or provide it as a companion branch/PR so you have the live API surface to bind to immediately.

2. **Schema Alignment**: The schema in PR #1218 covers:
   - Mount specifications & flange distances
   - Camera models with physical sensor dimensions (mm) and crop factors
   - Lens profiles with coverage flags, focal ranges, and mount compatibility
   If there are specific fields, enum keys, or structures needed for your UI components (e.g. display name helpers, group keys, or Lensfun IDs), let us know and we can adjust before locking it down.

3. **UI Scope & Bounty Split**:
   - **Our Scope**: Canonical camera/lens database schema, backend loader/parser, mount compatibility resolution engine, video EXIF metadata matching, and test suite.
   - **Your Scope**: QML/Qt UI selector components (`LensCalibrate.qml`, `LensProfile.qml`), wiring selectors to pre-populate from video metadata, and calibration validation guards.
   - **Split**: A 50/50 split ($250 / $250) of the $500 Algora bounty reflects a clean division of labor between backend database/loader and frontend UI/validation.

@AdrianEddy Could you please review and confirm if this scope split works for the project roadmap, and if you would be open to approving `/split @ElvinGts $250 @snchngny $250` upon completion?

**Comment by Nabeel-javed** [comment]
Hi @AdrianEddy — I’m interested in contributing a focused part of this refactor. I see the complete implementation proposed in #1212 and the database/UI split being discussed around #1218. Is there any additional scope you would like a new contributor to take on, such as a specific remaining integration or regression-testing gap?

Before I start, could you confirm the acceptance criteria, whether that scope has a separate share of the advertised bounty, and how that share would be approved through Algora? I would use AI-assisted development and provide executed checks and a demo where applicable. If the existing submissions cover the funded work, please let me know and I’ll leave it with those contributors.

**Comment by ElvinGts** [comment]
Hi @Nabeel-javed! Thanks for your interest in this refactor! 👋

@snchngny and I have already started coordinating a scope split on this issue:
- **My scope (PR #1218):** Canonical camera/lens database schema (`camera_lens_db.json` v2.5.0), backend loader/parser, mount compatibility engine, video EXIF metadata matching, and test suite.
- **@snchngny's scope:** QML/Qt UI selector components, calibration validation guards, and wiring selectors to pre-populate from video metadata.

We're waiting for @AdrianEddy to confirm the `/split` arrangement. If there's a third subtask that fits your expertise (e.g., Lensfun integration, additional calibration profiles, or documentation), we'd love to have you on board! Feel free to propose what area you'd like to take on.

**Comment by Nabeel-javed** [comment]
Thanks @ElvinGts. I checked the existing submissions before choosing a scope: #1212 already includes Lensfun catalogue metadata and a companion lens_profiles PR, and @AdrianEddy has asked for no additional PRs on #150. A separate Lensfun implementation would overlap that work.

I can offer a focused integration/regression pass against the implementation @AdrianEddy selects: check offline catalogue loading and catalogue updates, verify that Lensfun-only selections cannot be used as calibrated distortion profiles, and exercise camera/lens selection through metadata prefill and upload validation. Deliverables would be a reproducible results matrix and regression tests or fixes for any confirmed gaps, submitted against the selected base. This would use AI-assisted development with executed checks.

@AdrianEddy, would that be useful as a separately accepted $200 scope? I see the proposed $250/$250 split already covers the advertised $500, so I’m not assuming an additional share is available. Please confirm the selected base, acceptance criteria and allocation through Algora before I start. If this testing is already covered or no budget remains, I’ll leave it with the existing contributors.

**Comment by liveeeeee** [comment]
/attempt #742

### Implementation Plan
1. **Canonical Camera & Lens Registry (camera_registry.rs + 
esources/camera_registry.json)**:
   - Aggregate & normalize camera brands, models, lens mounts, sensor dimensions, and crop factors across existing Gyroflow profiles and LensFun XML database.
   - Provide clean companion generator script _scripts/generate_camera_registry.py to maintain reproducible updates without bloating Git history.
2. **Compatibility & Recommendation Engine**:
   - CameraRegistry::get_compatible_cameras() matching mount type and crop factor compatibility (e.g. Sony full-frame E-mount cross-model recommendations).
3. **UI Selectors & Video Metadata Prefill**:
   - Modernize LensProfile.qml and LensCalibrate.qml with hierarchical brand/model/lens selectors, retaining clear "Other" manual input fallback.
   - Pre-populate selectors from video telemetry/EXIF metadata on video load.
4. **Review & Bad Profile Management**:
   - Profile review toggle for same camera/lens setup with decoupled local-hide persistence vs. community rating.
5. **Calibrator Integrity Validation**:
   - Prevent submitting incomplete/ambiguous lens profiles (e.g. missing lens parameters on interchangeable lens cameras, zoom lenses without focal length).

Validation: cargo unit tests for registry loading & compatibility checks, QML syntax checks, and demo walkthrough.

**Comment by liveeeeee** [comment]
/attempt #742

### Implementation Plan
1. Refactored lens profile handling with canonical camera and lens profile registry across 8 files (`+2679, -4`).
2. Integrated brand, camera, and lens model selectors with pre-population from video metadata.
3. Added metadata validation and review/upload mechanism.
4. CLA signed, zero merge conflicts, all tests passing in PR #1221.

**Comment by ItsJazii** [comment]
/attempt #742

I'd like to work on the lens profile refactor.

**Background:** I ship desktop apps (Pane — Rust/Tauri, winget-published: https://github.com/ItsJazii/pane). Pane stays portfolio-only; I won't change that repo for this bounty.

**Plan:**
1. Inventory current lens profile strings + LensFun data; propose a normalized JSON schema for brands / models / lenses / mounts / sensor sizes / crop factor under `resources`, with the same auto-update path as existing lens profiles (and mirror into https://github.com/gyroflow/lens_profiles).
2. Replace free-text search with cascading selectors (brand → model → lens), prefilled from video metadata where possible; keep an "Other" + manual entry path.
3. Calibrator: require a clear setup before submit (block vague mirrorless-without-lens, zoom-without-focal-length, missing brand/model); wire LensFun into the same selectors (#150).
4. Add a review UI to toggle between submitted profiles for a given camera setup so bad ones are easy to spot/remove.
5. Open a draft PR early with schema + one selector path, then iterate on calibrator enforcement + review tooling.

Happy to adjust scope with @AdrianEddy if there's a preferred first slice.

**Comment by ItsJazii** [comment]
First-slice draft PR up (catalog + selectors; **no `/claim` yet**): https://github.com/gyroflow/gyroflow/pull/1226

Happy to iterate toward export-gating / review UI / lens_profiles publishing in follow-ups if this direction looks right.

**Comment by stffinfcti** [comment]
/attempt #742

Implementation plan (building on the approach maintainer endorsed in #1118, which currently has compile-blocking gaps):

- Ship a generated canonical camera/lens database at `resources/camera_database.json` (brands, models, lenses, mounts, sensor size, crop factor), produced by a reproducible `_scripts` generator from the gyroflow lens profile database + LensFun, and auto-updated alongside lens profiles.
- Rust `camera_database` module in core: brand/model/lens queries, compatible-camera lookup (shared mount + sensor class), and fuzzy matching of video-file metadata to known cameras for prefill.
- Lens profile search replaced by cascading brand/camera/lens selectors (still able to fall back to free search); calibrator uses the same selectors with "Other" manual entry.
- Review flow to toggle between all submitted profiles for a camera setup, with hide-bad handled locally.
- Submission validation: block uploads lacking a clear setup (no brand/model, interchangeable-lens camera without lens, zoom lens without focal length).

Working demo video + build/test receipt will be included in the PR per the bounty rules.

**Comment by themaksat** [comment]
I have submitted a solution for this issue in PR https://github.com/gyroflow/gyroflow/pull/1229. Full implementation and verification included. Payout wallet (Solana): `2BTCUUTviLNDtZDRBhPBTwSXZMrTttPJNbGevMXBBMKK`

**Comment by TheAliphant** [comment]
I can take a bounded first slice, pending maintainer confirmation of scope and the current funded payout route.

Proposed slice: add/validate the resource-backed camera/lens metadata model (brand, model, mount, sensor size, crop factor), preserve the existing lens-profile update path, and keep unknown values representable as `Other`. I would verify stable IDs and an update-safe merge against an existing profile before touching the full calibrator/UI selector rewrite.

This is intentionally not a claim on the entire multi-part issue. Please confirm whether this slice is wanted and what acceptance milestone/payout applies. I have a scope/acceptance proof map ready locally: `work-products/github-proof-gyroflow-742/acceptance-map.md`.

**Comment by northernautoparts** [comment]
Is the $500 bounty still available to new contributors? I see the existing PRs and proposed scope splits, so I would only start if there is a remaining scope you want covered. Which acceptance criteria, if any, remain unmet? I would provide executed tests and the required demo. If the funded work is already covered, no problem—I will leave it with the existing contributors.

**Comment by shashank03-dev** [comment]
/attempt #742

Hi @AdrianEddy, I'd like to take this on, but it's a big one so I want to split it into PRs you can actually review instead of one giant diff.

Proposed order:
1. Camera database: generate resources/camera_database.json (brand, model, mount, sensor size, crop factor) from the current lens_profiles repo plus lensfun, and add a CameraDatabase loader in core with tests. No UI changes yet.
2. Calibrator: require picking a camera from the list (with "Other" for manual entry) and block submissions missing key info like focal length for zoom lenses.
3. Main window: replace the search box with brand / model / lens selectors, prefilled from video metadata, with compatible camera suggestions.
4. Review mode to flip between all submitted profiles for one setup.

A few questions before I start:
- Should the camera database live in gyroflow/lens_profiles and auto-update like profiles do, or ship in resources first?
- Is a separate PR per step fine for the bounty, or do you want it paid out only at the end?

I'm working in Rust and Python and I use AI tooling as part of my workflow. Every change gets built and tested locally before I open anything.

**Comment by ntoledo319** [comment]
Hi @AdrianEddy — before starting work, could you confirm two things about this bounty?

1. **Is the $500 bounty still active and funded?** I see ~10 open PRs (#1154, #1167, #1212, #1218, #1221, #1229, and others) plus a proposed 50/50 split between @ElvinGts and @snchngny — none reviewed yet, so I want to avoid duplicating work you already consider covered.

2. **What are the acceptance criteria for a fresh submission?** Between the original checklist and your Dec 2025 clarification, the bar seems to be all 8 bullets against current master (catalog + selectors + metadata prefill + calibrator validation + review flow, with demo video). Is that still the target, or is there a specific remaining slice you'd rather a new contributor take on?

I'm ready to start immediately once you confirm. Transparency note: I use AI-assisted tooling in my workflow; everything would be built and tested locally with the required demo video in the PR per the bounty rules.

Thanks for the great project!
